### PR TITLE
enh(conf) remove obsolete engine parameters

### PR DESCRIPTION
**Pull request by a-launois**
## Description

remove obsolete engine parameters, update some default values

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 1]
```diff
@@ -344,18 +328,14 @@ public function export($filterName = null)
...
-                        || $parameter == 'host_perfdata_file_processing_command'
-                        || $parameter == 'service_perfdata_file_processing_command'
-                        || $parameter == 'ochp_command'
-                        || $parameter == 'ocsp_command'
                     ) {
                         $tmp = $this->commandObj->getParameters($value, $this->commandObj->getUniqueLabelField());
                         $value = $tmp[$this->commandObj->getUniqueLabelField()];
-                    } elseif ($parameter == 'illegal_object_name_chars'
+                    } elseif (
+                        $parameter == 'illegal_object_name_chars'
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:338]:
```suggestion
                        $parameter === 'illegal_object_name_chars'
```

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 2]
```diff
@@ -344,18 +328,14 @@ public function export($filterName = null)
...
                         || $parameter == 'illegal_macro_output_chars'
```
Comment (a-launois) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:339]:
```suggestion
                        || $parameter === 'illegal_macro_output_chars'
```

File: www/install/var/baseconf/centreon-engine.sql [hunk 3]
```diff
@@ -1,8 +1,89 @@
...
+INSERT INTO `cfg_nagios` (`nagios_id`, `nagios_name`) VALUES (1, 'Centreon Engine Central');
+UPDATE `cfg_nagios` SET `log_file` = '@monitoring_varlog@/centengine.log';
+UPDATE `cfg_nagios` SET `cfg_dir` = '@monitoring_etc@';
```
Comment (callapa) [inline www/install/var/baseconf/centreon-engine.sql:6]: Can you use a single UPDATE request, please?

File: www/include/configuration/configNagios/DB-Func.php [hunk 4]
```diff
@@ -84,13 +84,12 @@ function enableNagiosInDB($nagiosId = null)
...
 
-    $query = "SELECT `id`, `name` FROM nagios_server WHERE `ns_activate` = '0' " .
-             "AND `id` = :id";
+    $query = "SELECT `id`, `name` FROM nagios_server WHERE `ns_activate` = '0' AND `id` = :id";
     $statement = $pearDB->prepare($query);
     $statement->bindValue(':id', (int) $data["nagios_server_id"], \PDO::PARAM_INT);
     $statement->execute();
     $activate = $statement->fetch(\PDO::FETCH_ASSOC);
-    if ($activate && $activate["name"]) {
+    if (isset($activate["name"]) && $activate["name"] == 1) {
```
Comment (kduret) [inline www/include/configuration/configNagios/DB-Func.php:92]: `name` is the name of the poller, it is not a tinyint for activation

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 5]
```diff
@@ -329,137 +312,139 @@ public function insertServerInCfgNagios($source, $iId, $sName)
...
-            $res = $this->DB->execute($stmt, $params);
+
+            foreach ($params as $paramName => $paramValue) {
+                if ($paramName === ':nagios_server_id') {
+                    $stmt->bindValue($paramName, $paramValue, \PDO::PARAM_INT);
+                } else {
+                    $stmt->bindValue($paramName, empty($paramValue) ? null : $paramValue, \PDO::PARAM_STR);
+                }
+            }
+            $res = $stmt->execute();
```
Comment (kduret) [inline www/class/centreon-config/centreonMainCfg.class.php:447]:
```suggestion
            $stmt->execute();
```

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 6]
```diff
@@ -329,48 +312,43 @@ public function insertServerInCfgNagios($source, $iId, $sName)
...
+            `illegal_macro_output_chars`, `use_regexp_matching`, `use_true_regexp_matching`, `admin_email`,
+            `admin_pager`, `nagios_comment`, `nagios_activate`, `event_broker_options`,
+            `enable_predictive_host_dependency_checks`, `enable_predictive_service_dependency_checks`,
+            `passive_host_checks_are_soft`, `enable_environment_macros`, `debug_file`, `debug_level`,
+            `debug_level_opt`, `debug_verbosity`, `max_debug_file_size`, `cfg_file`
+            )
+            VALUES (
+            ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?,
+            ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?,
+            ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
```
[the remark this answers is not in the document]
Comment (callapa) [reply www/class/centreon-config/centreonMainCfg.class.php]: Yes indeed

**Pull request merged**